ext/dom/parentnode.c

```diff
@@ -255,10 +255,33 @@ static void dom_fragment_assign_parent_node(xmlNodePtr parentNode, xmlNodePtr fr
 	fragment->last = NULL;
 }
 
+static zend_result dom_hierarchy_node_list(xmlNodePtr parentNode, zval *nodes, int nodesc)
+{
+	for (int i = 0; i < nodesc; i++) {
+		if (Z_TYPE(nodes[i]) == IS_OBJECT) {
```

Comment:
Wait, can a node *not* be an object?

Reply:
Spec says you may insert strings, so it can be IS_STRING too. In that case we create a text node, but since that's a new object it cannot violate the dom hierarchy.

Reply:
Right, I didn't think that the zval node list could contain anything other than objects, as I would have expected free strings to always be encoded as `DOMText` object. Just having wrong assumptions.